### ISSUE TITLE
Make UI more keyboard-friendly

### ISSUE THREAD

**Issue by Atario** [erengy/taiga#82]
There are a lot of improvements that could be made in this area — keyboard ability is pretty abysmal at the moment.  Menus (!) seem to be 100% inaccessible by keyboard, and the rest of the UI is quite unfriendly to keyboard navigation as well.

**Comment by erengy** [comment]
Previously discussed in #46 and #51. It's not *that* bad, though. Anyway, I've been meaning to open an issue for this myself, so here's that:

- [x] Anime Information: Use <kbd>Tab</kbd> and <kbd>Shift</kbd>+<kbd>Tab</kbd> to navigate dialog items
- [x] Anime Information: Use <kbd>Ctrl</kbd>+<kbd>Tab</kbd> and <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>Tab</kbd> to navigate tabs
- [x] Main: Use <kbd>Tab</kbd> and <kbd>Shift</kbd>+<kbd>Tab</kbd> to navigate dialog items
- [x] Main: Use <kbd>Ctrl</kbd>+<kbd>Tab</kbd> and <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>Tab</kbd> to navigate tabs
- [ ] Main: Access menu bar via <kbd>Alt</kbd> and <kbd>F10</kbd>
- [x] Main: Access search bar via <kbd>Ctrl</kbd>+<kbd>F</kbd> and <kbd>F3</kbd>
- [ ] Main: Fix bug where user can't navigate up above separator lines on the side panel
- [ ] Main: Fix bug where keyboard focus is stolen by the side panel while switching Anime List tabs with Ctrl+Tab

**Comment by erengy** [comment]
I've finished implementing all the features above; they're going to be available in v1.2. The UI is probably still not entirely usable without a mouse, but it should be a lot more keyboard-friendly now.

Moving on: I'd like to set up shortcut keys for some commonly used features, but I'm not sure what key combinations would be the most appropriate (they're not going to be customizable, at least for now). What I have in mind is as follows:

Action|Shortcut
------|--------
Open library folders|<kbd>*modifier*</kbd>+<kbd>1</kbd>, <kbd>*modifier*</kbd>+<kbd>2</kbd>, *(...)*
Play next episode (of last watched anime)|?
Play random episode|?
Scan available episodes|? (currently <kbd>F5</kbd>)
Synchronize list|?
Support|<kbd>F1</kbd>

**Comment by shortcake** [comment]
"Close to tray" has stopped working like it used to in the stable versions.
Clicking the Close button at the top-right corner still works, but pressing Alt-F4 exits the program instead of closing the window.

**Comment by erengy** [comment]
This was an intentional change. I thought it'd be more appropriate for <kbd>Alt</kbd>+<kbd>F4</kbd> to exit the application, even if "Close to tray" option is enabled. For the same reason, I set it as the shortcut key for *File → Exit*.

Apparently, people have varying opinions on this (you must be shocked right now :). [Here](https://discussions.agilebits.com/discussion/28124/option-to-minimize-to-tray-using-alt-f4) is a similar discussion, which ends up with the change being reverted. [Here](http://superuser.com/questions/431303/is-there-a-keyboard-shortcut-to-end-an-application-more-than-just-altf4), on the other hand, people think that doing so is inconsistent.

I don't have a strong preference on this. It's quite easy to simply remove the shortcut, or assign another one such as <kbd>Ctrl</kbd>+<kbd>Q</kbd> for the same purpose. I'll be releasing v1.2 today, let's see if anyone else will be bothered by it.

**Comment by shortcake** [comment]
Thank you for your response.

I can certainly understand wanting to end an application using Alt+F4. But that doesn't work with how I use my background applications.

There are several programs that are always running in my notification area: Taiga, Steam, gaming peripheral software, sound/video card software. When clicked, a window appears. When Alt+F4 is pressed, the window closes but the program keeps running. Taiga is now the only program I always have running that doesn't behave like this.

Also, I think it's inconsistent that Alt+F4 and the Close button have different functions. I don't think I've ever used another program that does this. Shouldn't they do the same thing?

**Comment by rr-** [comment]
I agree that <kbd>Alt</kbd>+<kbd>F4</kbd> should definitely minimize to tray if that option is enabled.
If I really want to quit, I should hit <kbd>Alt</kbd>, <kbd>&uarr;</kbd> (to focus "exit") and <kbd>Enter</kbd> to confirm that.

**Comment by rr-** [comment]
:+1:

**Comment by ConnorKrammer** [comment]
This is... a bit of a grey area, UX-wise. Clicking the close button is an obviously well-defined behaviour: if "close to tray" is true, close to tray, else close the window. With <kbd>Alt</kbd>+<kbd>F4</kbd>, however, it is up to the application to decide what behaviour to handle; there are applications that do it one way and applications that do it another. [This might be good to look at](https://askubuntu.com/questions/46917/what-is-the-purpose-of-alt-f4), even though it technically pertains to Ubuntu. Most interestingly, it mentions <kbd>Alt</kbd>+<kbd>F4</kbd> *defaulting* to being equivalent to hitting the close button. **Based on this, I would be inclined to make them perfectly analogous**, even if "close to tray" is enabled. (And I see that the latest commit does in fact do this.)

For a convenient way to quit the application regardless of the other settings, I would propose <kbd>Alt</kbd>+<kbd>Q</kbd>, which is a fairly established convention. It definitely beats having to find the Taiga icon in the tray, right click, and hit "exit".

Since you're holding off on making these shortcuts configurable, it's important that at least something hard-quits the application. That at least allows custom solutions like AutoHotkey to remap things in the interim.

**Comment by Zarggg** [comment]
While that may apply for the Ubuntu user base, I will add that my own experience with Windows tends to assume *both* <kbd>Alt</kbd>+<kbd>F4</kbd> and <kbd>Ctrl</kbd>+<kbd>Q</kbd> are used as "hard quit" options, with favor being given to <kbd>Alt</kbd>+<kbd>F4</kbd> by convention.

Since most of my early computer experience was on Macintosh computers (where the keyboard shortcut was <kbd>⌘</kbd>+<kbd>Q</kbd>), using <kbd>Alt</kbd>+<kbd>Q</kbd> as the default "hard quit" shortcut and having <kbd>Alt</kbd>+<kbd>F4</kbd> follow the "close to tray" option of the close button would seem more intuitive to me.

**Comment by ConnorKrammer** [comment]
That’s actually what I was suggesting – sorry if I was unclear. <kbd>Ctrl</kbd>+<kbd>Q</kbd> would always be “hard quit”, while <kbd>Alt</kbd>+<kbd>F4</kbd> would do the exact same thing as the close button – close to tray if enabled, else exit.

**Comment by rr-** [comment]
I think it's worth mentioning that if I have "close to tray" enabled and if Alt+F4 hard quits the application, I can't minimize it to tray unless I resort to using mouse (= very bad thing for me). Even if there is some workaround, I don't know it, and IMHO that's a valuable user story here.

**Comment by Atario** [comment]
Within Windows:

* <kbd>Alt+F4</kbd> is supposed to be the exact same thing as the close-window button, as is <kbd>Alt+Space, C</kbd> (Control menu, Close).  It's not "hard quit" unless you're on the application's only window and it is not configured to do something else in that case (e.g., "close to tray").
* <kbd>Alt+Q</kbd> is nothing standard.
* <kbd>Alt+F, X</kbd>  (File, Exit) is more usually known as a "hard quit".
* <kbd>Alt+Space, N</kbd> (Control menu, Minimize) is the same thing as the minimize button.

All of the above is the case in, for example, Firefox and Chrome.
